apps/file/utils.py

```diff
@@ -0,0 +1,52 @@
+import os
+import pandas as pd
+from django.db import transaction
+
+from apps.file.models import File
+from apps.core.models import Dataset, Table
+from utils.common import get_file_extension
+
+
+def process_excel_file(dataset: Dataset):
+    pd_excel = pd.ExcelFile(dataset.file.file.path)
+    sheets = [sheet for sheet in pd_excel.sheet_names]
+    with transaction.atomic():
+        for sheet in sheets:
+            table_data = {
+                "dataset": dataset,
+                "name": sheet,
+                "created_by": dataset.file.created_by,
+                "modified_by": dataset.file.modified_by,
+            }
+            Table.objects.create(**table_data)
+
+
+def process_csv_file(dataset: Dataset):
+    table_data = {
+        "dataset": dataset,
+        "name": os.path.basename(dataset.file.file.name),
+        "created_by": dataset.file.created_by,
+        "modified_by": dataset.file.modified_by,
+    }
+    Table.objects.create(**table_data)
+
+
+def create_dataset_and_tables(file: File) -> Dataset:
+    dataset_data = {
+        "created_by": file.created_by,
+        "modified_by": file.modified_by,
+        "name": os.path.basename(file.file.name),
+        "file": file,
+    }
+    dataset = Dataset.objects.create(**dataset_data)
+    # Now process file and create associated table
+    extension = get_file_extension(file.file.name)
+
+    if extension.lower() == "xlsx":
+        process_excel_file(dataset)
+    elif extension.lower() == "csv":
+        process_csv_file(dataset)
+    else:
+        # TODO: Handle gracefully, or should we? because serializer already validates extension
+        raise Exception("Invalid file type")
```

Comment:
Do we need to handle this here since serializer would have already handle this?

Reply:
No, but just in case if this is called from elsewhere.